projects/core/src/cms/store/effects/page.effect.ts

```diff
@@ -11,81 +13,65 @@ import {
   take
 } from 'rxjs/operators';
 
-import * as pageActions from '../actions/page.action';
 import * as componentActions from '../actions/component.action';
-import { OccCmsService } from '../../occ/occ-cms.service';
-import { DefaultPageService } from '../../services/default-page.service';
-
-import { Page } from '../../model/page.model';
+import * as pageActions from '../actions/page.action';
 import { ContentSlotData } from '../../model/content-slot-data.model';
-
-import { RoutingService, PageContext } from '../../../routing/index';
-import { PageType, CMSPage } from '../../../occ/occ-models/index';
+import { Page } from '../../model/page.model';
+import { OccCmsService } from '../../occ/occ-cms.service';
+import { RoutingService } from '../../../routing/index';
+import { CMSPage, CmsComponent } from '../../../occ/occ-models/index';
 
 @Injectable()
 export class PageEffects {
   @Effect()
-  loadPage$: Observable<any> = this.actions$.pipe(
-    ofType(
-      pageActions.LOAD_PAGEDATA,
-      pageActions.REFRESH_LATEST_PAGE,
-      '[Site-context] Language Change',
-      '[Auth] Logout',
-      '[Auth] Login'
-    ),
-    map((action: pageActions.LoadPageData) => action.payload),
-    switchMap(pageContext => {
-      if (pageContext === undefined) {
-        return this.routingService.getRouterState().pipe(
-          filter(routerState => routerState && routerState.state),
-          filter(routerState => routerState.state.cmsRequired),
-          map(routerState => routerState.state.context),
-          take(1),
-          mergeMap(context =>
-            this.occCmsService.loadPageData(context).pipe(
-              mergeMap(data => {
-                return [
-                  new pageActions.LoadPageDataSuccess(
-                    this.getPageData(data, context)
-                  ),
-                  new componentActions.GetComponentFromPage(
-                    this.getComponents(data)
-                  )
-                ];
-              }),
-              catchError(error => of(new pageActions.LoadPageDataFail(error)))
-            )
+  refreshPage$: Observable<Action> = this.actions$.pipe(
+    ofType('[Site-context] Language Change', '[Auth] Logout', '[Auth] Login'),
+    // TODO:#1135v2 - switchMap?
+    switchMap(_ =>
+      this.routingService.getRouterState().pipe(
+        filter(routerState => routerState && routerState.state),
```

Comment:
we can combine these 2 filters into 1.

Reply:
Ok, good point

Reply:
we need it. Refresh cms page data only when page needs cms. For us, there is no pages NOT use cms. But, customers may make some pages not use cms.